sdk/kiosk/src/utils.ts

```diff
@@ -81,37 +86,82 @@ export function extractKioskData(data: DynamicFieldInfo[]): KioskData {
           acc.items.push({
             itemId: val.objectId,
             itemType: val.objectType,
+            isLocked: false,
           });
           break;
         case 'kiosk::Listing':
           acc.listingIds.push(val.objectId);
-          acc.listings.push({
+          listings.push({
             itemId: val.name.value.id,
             listingId: val.objectId,
             isExclusive: val.name.value.is_exclusive,
           });
           break;
+        case 'kiosk::Lock':
+          lockedItemIds?.push(val.name.value.id);
+          break;
       }
       return acc;
     },
-    { listings: [], items: [], itemIds: [], listingIds: [] },
+    { items: [], itemIds: [], listingIds: [], extensions: [] },
   );
 }
 
-// simple multiGetObjects wrapper to simplify cases on functions.
-export function getObjects(
-  provider: JsonRpcProvider,
-  ids: string[],
-  options: SuiObjectDataOptions,
-) {
-  if (ids.length === 0) {
-    return Promise.resolve([]);
-  }
-
-  return provider.multiGetObjects({ ids, options });
-}
-
 // e.g. 0x2::kiosk::Item -> kiosk::Item
 export const getTypeWithoutPackageAddress = (type: string) => {
   return type.split('::').slice(-2).join('::');
 };
+
+/**
+ * A helper that attaches the listing prices to kiosk listings.
+ */
+export const attachListingsAndPrices = (
+  kioskData: KioskData,
+  listings: KioskListing[],
+  listingObjects: SuiObjectResponse[],
+) => {
+  // map item listings as {item_id: KioskListing}
+  // for easier mapping on the nex
+  const itemListings = listings.reduce<Record<string, KioskListing>>(
+    (acc: Record<string, KioskListing>, item, idx) => {
+      acc[item.itemId] = { ...item };
+
+      // return in case we don't have any listing objects.
+      // that's the case when we don't have the `listingPrices` included.
+      if (listingObjects.length === 0) return acc;
+
+      const data = getObjectFields(listingObjects[idx]);
+      if (!data) return acc;
+
+      acc[item.itemId].price = data.value;
+      return acc;
+    },
+    {},
+  );
+
+  kioskData.items.map((item) => {
+    item.listing = itemListings[item.itemId] || undefined;
+  });
+};
+
+/**
+ * A Helper to attach locked state to items in Kiosk Data.
+ */
+export const attachLockedItems = (
```

Comment:
Are you sure this need to be separate functions? The logic is pretty straightforward and this function is not reused.

Reply:
Added it to make the `fetchKiosk` function easier to read (not going through so many lines of code all at once)